modules/dnn/test/test_layers.cpp

```diff
@@ -618,6 +618,179 @@ TEST(Layer_LSTM_Test_Accuracy_with_, HiddenParams)
+
+    std::vector<Mat> inputs{input};
+    std::vector<Mat> outputs;
+
+    runLayer(layer, inputs, outputs);
+    ASSERT_EQ(shape(output_ref), shape(outputs[0]));
+    normAssert(output_ref, outputs[0]);
+}
+
+INSTANTIATE_TEST_CASE_P(/*nothing*/, Layer_ARG_1d_Test, Combine(
+/*input blob shape*/    Values(0, 1, 2, 3),
+/*operation*/           Values(0, 1),
+/*operation*/           Values( "max", "min")
+));
+
+typedef testing::TestWithParam<tuple<int, std::string>> Layer_NARYELEMWISE_1d_Test;
+TEST_P(Layer_NARYELEMWISE_1d_Test, Accuracy) {
+
+    int batch_size = get<0>(GetParam());
+    std::string operation = get<1>(GetParam());
+
+    LayerParams lp;
+    lp.type = "Eltwise";
+    lp.name = "addLayer";
+    lp.set("operation", operation);
+    Ptr<NaryEltwiseLayer> layer = NaryEltwiseLayer::create(lp);
+
+    std::vector<int> input_shape = {batch_size, 1};
+    if (batch_size == 0)
+        input_shape.erase(input_shape.begin());
+
+    cv::Mat input1 = cv::Mat(input_shape, CV_32F, 2.0);
+    cv::Mat input2 = cv::Mat(input_shape, CV_32F, 1.0);
+
+    // dynamicly select the operation
+    cv::Mat output_ref = (operation == "sum") ? input1 + input2 :
+                         (operation == "mul") ? input1.mul(input2) :
+                         (operation == "div") ? input1 / input2 : input1 - input2;
+
+    std::vector<Mat> inputs{input1, input2};
+    std::vector<Mat> outputs;
+
+    runLayer(layer, inputs, outputs);
+    ASSERT_EQ(shape(output_ref), shape(outputs[0]));
+    normAssert(output_ref, outputs[0]);
+}
+
+INSTANTIATE_TEST_CASE_P(/*nothing*/, Layer_NARYELEMWISE_1d_Test, Combine(
+/*input blob shape*/    Values(0, 1),
+/*operation*/           Values("div", "mul", "sum", "sub")
+));
+
+typedef testing::TestWithParam<tuple<int, std::string>> Layer_ELEMWISE_1d_Test;
+TEST_P(Layer_ELEMWISE_1d_Test, Accuracy) {
+
+    int batch_size = get<0>(GetParam());
+    std::string operation = get<1>(GetParam());
+
+    LayerParams lp;
+    lp.type = "Eltwise";
+    lp.name = "addLayer";
+    lp.set("operation", operation);
+    Ptr<EltwiseLayer> layer = EltwiseLayer::create(lp);
+
+    std::vector<int> input_shape = {batch_size, 1};
+    if (batch_size == 0)
+        input_shape.erase(input_shape.begin());
+
+    cv::Mat input1 = cv::Mat(input_shape, CV_32F, 1.0);
+    cv::Mat input2 = cv::Mat(input_shape, CV_32F, 1.0);
+
+    // fill in the input matrices with different values
+    for (int i = 0; i < batch_size; ++i)
+    {
+        input1.at<float>(i, 0) = static_cast<float>(i);
+        input2.at<float>(i, 0) = static_cast<float>(i + 1);
+    }
+
+    // dynamicly select the operation
+    cv::Mat output_ref = (operation == "sum") ? input1 + input2 :
+                         (operation == "max") ? cv::max(input1, input2) :
+                         (operation == "min") ? cv::min(input1, input2) :
+                         (operation == "prod")  ? input1.mul(input2) : input1 / input2;
```

Comment:
I propose to add "div" check and stay the matrix empty, if operation was not matched. It highlights errors and incorrect instantiation of test.